src/components/sidebar/sidebar.pw.tsx

```diff
@@ -0,0 +1,441 @@
+import { expect, test } from "@playwright/experimental-ct-react17";
+import React from "react";
+import {
+  backgroundUILocator,
+  closeIconButton,
+  getComponent,
+  getDataElementByValue,
+} from "../../../playwright/components";
+import { CLOSE_ICON_BUTTON } from "../../../playwright/components/locators";
+import {
+  sidebarComponent,
+  sidebarPreview,
+} from "../../../playwright/components/sidebar";
+import { CHARACTERS } from "../../../playwright/support/constants";
+import {
+  assertCssValueIsApproximately,
+  checkAccessibility,
+  continuePressingSHIFTTAB,
+  continuePressingTAB,
+  waitForAnimationEnd,
+} from "../../../playwright/support/helper";
+import {
+  SidebarBackgroundScrollTestComponent,
+  SidebarBackgroundScrollWithOtherFocusableContainers,
+  SidebarComponent,
+  SidebarComponentFocusable,
+  SidebarComponentWithHeading,
+  SidebarComponentWithOnCancel,
+  TopModalOverride,
+} from "./components.test-pw";
+import { SidebarProps } from "./sidebar.component";
+import { SIDEBAR_SIZES, SIDEBAR_SIZES_CSS } from "./sidebar.config";
+
+test.describe("Prop tests for Sidebar component", () => {
+  ([true, false] as const).forEach((boolVal) => {
+    test(`check enableBackgroundUI prop set to ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<SidebarComponent enableBackgroundUI={boolVal} />);
+
+      const backgroundUILocatorElement = backgroundUILocator(page);
+
+      if (boolVal) {
+        await expect(backgroundUILocatorElement).not.toBeVisible();
+      } else {
+        await expect(backgroundUILocatorElement).toBeVisible();
+      }
+    });
+  });
+
+  ([
+    ["left", 0, 852],
+    ["right", 852, 0],
+  ] as [SidebarProps["position"], number, number][]).forEach(
+    ([boolVal, left, right]) => {
```

Comment:
```suggestion
    ([position, left, right]) => {
``` just the variable name "boolVal" sounds like it should be either true/false